### DISCUSSION TITLE
Custom Layouts (via Search Extensibility Library) for updated version 4.7 and @pnp/modern-search-extensibility 1.3.0

### DISCUSSION THREAD

**Discussion by bergersj** [microsoft-search/pnp-modern-search#2273] [Q&A]
Has anyone been able to update a search extensibility library to work with the updated PnP search results webpart after it was updated from 4.6.1 to 4.7? 

It doesn't appear the documentation has been updated yet, [here](https://microsoft-search.github.io/pnp-modern-search/extensibility/custom_layout/)

Similar to #2205 

I have added the renderType property to the layout definition and specified Handlerbars, but now I'm getting serviceKey errors that don't quite make sense to me and that I wasn't getting prior to updating @pnp/modern-search-extensibility

@wobba @FranckyC 

`Error - [tsc] src/libraries/searchExt/SearchExtLibrary.ts(27,110): error TS2345: Argument of type 'typeof SearchSuggestionProvider' is not assignable to parameter of type 'new (serviceScope: ServiceScope) => ISuggestionProvider'.
Error - [tsc] src/libraries/searchExt/SearchExtLibrary.ts(55,90): error TS2345: Argument of type 'typeof EventsLayout' is not assignable to parameter of type 'new (serviceScope: ServiceScope) => ILayout'.
Error - [tsc] src/libraries/searchExt/SearchExtLibrary.ts(64,89): error TS2345: Argument of type 'typeof NewsLayout' is not assignable to parameter of type 'new (serviceScope: ServiceScope) => ILayout'.`

Dependencies:
```
    "dependencies": {
        "@microsoft/sp-component-base": "1.12.1",
        "@microsoft/sp-core-library": "1.12.1",
        "@microsoft/sp-http": "1.12.1",
        "@microsoft/sp-lodash-subset": "1.12.1",
        "@microsoft/sp-property-pane": "1.12.1",
        "@microsoft/sp-webpart-base": "1.12.1",
        "@pnp/modern-search-extensibility": "1.3.0",
        "@pnp/pnpjs": "2.0.13",
        "@webcomponents/custom-elements": "1.4.3",
        "@webcomponents/webcomponentsjs": "2.5.0",
        "datebook": "^7.0.8",
        "handlebars": "4.7.7",
        "lzutf8": "^0.6.1",
        "moment": "^2.29.1",
        "office-ui-fabric-react": "6.214.0",
        "react": "16.8.5",
        "react-clamped": "^1.0.1",
        "react-dom": "16.8.5"
    }
```

**Comment by FranckyC** [comment]
Hi @bergersj, make sure you are using the same exact npm package versions as the core Web Parts (SPFx version + pnp extensibility library). For @pnp/modern-search-extensibility, try to use a strict version 1.3.0, not the latest one.

**Reply by Fringie** [reply]
Hey dude,

I'm running into the same issue. I'm going to have a play around with it, but can you advise what versions each of these should be?

```
"dependencies": {
    "@microsoft/sp-component-base": "^1.16.0",
    "@microsoft/sp-core-library": "^1.16.0",
    "@microsoft/sp-http": "^1.16.0",
    "@microsoft/sp-page-context": "^1.16.0",
    "@microsoft/sp-webpart-base": "^1.16.0",
    "@pnp/modern-search-extensibility": "^1.5.0",
    "tslib": "2.3.1"
  },
```

**Comment by wobba** [comment]
1.14 was used for v4.7
